wgengine/userspace.go

```diff
@@ -1373,6 +1436,19 @@ func (e *userspaceEngine) setTSMPPongCallback(data [8]byte, cb func(packet.TSMPP
 	}
 }
 
+func (e *userspaceEngine) setICMPEchoCallback(idSeq uint32, cb func()) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	if e.icmpEchoCallback == nil {
+		e.icmpEchoCallback = map[uint32]func(){}
+	}
```

Comment:
you can drop this init now if you use the new helpers we added recently:

```go
        if cb == nil {
		delete(e.icmpEchoCallback, idSeq)
	} else {
		mak.Set(&e.icmpEchoCallback, idSeq, cb)
	}
}
```

Your call. (it's safe to delete from a nil map too)

Reply:
Neat! Generics!